programs/stake/src/stake_state.rs

```diff
@@ -407,37 +407,45 @@ impl Stake {
     ///   for credits_observed were the points paid
     pub fn calculate_points_and_credits(
         &self,
-        vote_state: &VoteState,
+        new_vote_state: &VoteState,
         stake_history: Option<&StakeHistory>,
     ) -> (u128, u64) {
-        if self.credits_observed >= vote_state.credits() {
+        // if there is no newer credits since observed, return no point
+        if new_vote_state.credits() <= self.credits_observed {
             return (0, 0);
         }
 
-        let mut credits_observed = self.credits_observed;
-        let mut points = 0u128;
-        for (epoch, credits, prev_credits) in vote_state.epoch_credits() {
+        let mut points = 0;
+        let mut new_credits_observed = self.credits_observed;
```

Comment:
this is subtle, but let's match with the rhythm here: we're using (point, credits) tuple across this fn

Reply:
also,  the intention of `point` (without `new_`) and `new_credits_observed` (with `new_`) is that point is isolated and independent for each inflation distribution. So, it's stateless. But `credits_observed` is continuously updated thing in stake accounts. So, it's state-full. So, only `credits_observed` can be said there is newer version and (current/older) version.